File: counting_and_sequencing/source_code/lib/OneToTenPage.kt

class OneToTenPage extends StatelessWidget {
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            backgroundColor: Colors.purple[100],
        appBar: AppBar(
        title: Text('LEVEL 1'),
        centerTitle: true,
        backgroundColor: Colors.purple,
        ),
        body: Center(
        child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
        SizedBox(height: 20),
        Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        children: [
        NonClickableNumber(text: 'Two'),
        NonClickableNumber(text: 'Three'),
        NonClickableNumber(text: '______'),
        NonClickableNumber(text: 'Five'),
        ],
        ),
        SizedBox(height: 32),
        Text(
            'Choose an Option',
            style: TextStyle(
                    fontSize: 24,
        fontWeight: FontWeight.bold,
        color: Colors.white,
        ),
        ),
        SizedBox(height: 32),
        NumberOptionButton(
            text: 'One',
        onPressed: () => navigate(context, false),
        ),
        NumberOptionButton(
            text: 'Seven',
        onPressed: () => navigate(context, false),
        ),
        NumberOptionButton(
            text: 'Six',
        onPressed: () => navigate(context, false),
        ),
        NumberOptionButton(
            text: 'Four',
        onPressed: () => navigate(context, true),
        ),
        ],
        ),
        ),
        );
    }

    void navigate(BuildContext context, bool isCorrect) {
        if (isCorrect) {
            Navigator.push(
                context,
                MaterialPageRoute(
                    builder: (context) => RightAnswerPage(
                    onNextQuestionPressed: () {
                // Navigate to the next question in the sequence
                Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => TwentyThreeToTwentySixPage()),
                );
            },
            ),
            ),
            );
        } else {
            Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => WrongAnswerPage()),
            );
        }
    }
}

class TwentyThreeToTwentySixPage extends StatelessWidget {
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            backgroundColor: Colors.purple[100],
        appBar: AppBar(
        title: Text('LEVEL 1'),
        centerTitle: true,
        backgroundColor: Colors.purple,
        ),
        body: Center(
        child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
        SizedBox(height: 20),
        Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        children: [
        NonClickableNumber(text: 'Twenty Three'),
        NonClickableNumber(text: 'Twenty Four'),
        NonClickableNumber(text: 'Twenty Five'),
        NonClickableNumber(text: '____________'),
        ],
        ),
        SizedBox(height: 32),
        Text(
            'Choose an Option',
            style: TextStyle(
                    fontSize: 24,
        fontWeight: FontWeight.bold,
        color: Colors.white,
        ),
        ),
        SizedBox(height: 32),
        NumberOptionButton(
            text: 'Twenty Seven',
        onPressed: () => navigate(context, false),
        ),
        NumberOptionButton(
            text: 'Thirty Seven',
        onPressed: () => navigate(context, false),
        ),
        NumberOptionButton(
            text: 'Twenty Six',
        onPressed: () => navigate(context, true),
        ),
        NumberOptionButton(
            text: 'Twenty Two',
        onPressed: () => navigate(context, false),
        ),
        ],
        ),
        ),
        );
    }

    void navigate(BuildContext context, bool isCorrect) {
        if (isCorrect) {
            Navigator.push(
                context,
                MaterialPageRoute(
                    builder: (context) => RightAnswerPage(
                    onNextQuestionPressed: () { /*
              // Navigate to the next question in the sequence
              Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => TwentyThreeToTwentySixPage()),
              ); */
            },
            ),
            ),
            );
        } else {
            Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => WrongAnswerPage()),
            );
        }

    }
}

class NonClickableNumber extends StatelessWidget {
    final String text;

    NonClickableNumber({required this.text});

    @override
    Widget build(BuildContext context) {
        return Container(
            constraints: BoxConstraints(minWidth: 100),
        alignment: Alignment.center,
        padding: EdgeInsets.all(8),
        decoration: BoxDecoration(
        color: Colors.purple[50],
        border: Border.all(),
        borderRadius: BorderRadius.circular(15),
        ),
        child: Text(
        text,
        style: TextStyle(
        fontSize: 25,
        fontWeight: FontWeight.bold,
        color: Colors.black87,
        ),
        ),
        );
    }
}

class NumberOptionButton extends StatelessWidget {
    final String text;
    final VoidCallback onPressed;

    NumberOptionButton({required this.text, required this.onPressed});

    @override
    Widget build(BuildContext context) {
        return Padding(
            padding: const EdgeInsets.symmetric(vertical: 10.0),
        child: MouseRegion(
        cursor: SystemMouseCursors.click,
        child: ElevatedButton(
        onPressed: onPressed,
        child: Text(
        text,
        style: TextStyle(
        fontSize: 20,
        fontWeight: FontWeight.bold,
        ),
        ),
        style: ButtonStyle(
        backgroundColor: MaterialStateProperty.resolveWith<Color>(
        (Set<MaterialState> states) {
            if (states.contains(MaterialState.hovered)) {
                return Colors.deepPurpleAccent;
            }
            return Colors.deepPurple;
        },
        ),
        fixedSize: MaterialStateProperty.all(Size(180, 55)),
        shape: MaterialStateProperty.all<RoundedRectangleBorder>(
        RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(30),
        ),
        ),
        ),
        ),
        ),
        );
    }
}

class RightAnswerPage extends StatelessWidget {
    final VoidCallback onNextQuestionPressed;

    RightAnswerPage({required this.onNextQuestionPressed});

    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
                    title: Text('Right Answer'),
        ),
        body: Center(
        child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
        Text(
            'Congratulations!',
            style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
        ),
        SizedBox(height: 20),
        Text(
            'You answered correctly.',
            style: TextStyle(fontSize: 20),
        ),
        SizedBox(height: 20),
        ElevatedButton(
            onPressed: onNextQuestionPressed,
            child: Text(
                    'Next Question',
            style: TextStyle(fontSize: 18), // Adjust the font size as needed
        ),
        style: ElevatedButton.styleFrom(
        padding: EdgeInsets.symmetric(horizontal: 20, vertical: 15), // Adjust the padding as needed
        ),
        ),
        ],
        ),
        ),
        );
    }
}


class WrongAnswerPage extends StatelessWidget {
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
                    title: Text('Wrong Answer'),
        ),
        body: Center(
        child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
        Text('Oops! Your answer is incorrect.',
            style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
        ),
        SizedBox(height: 20),
        ElevatedButton(
            onPressed: () {
            Navigator.pop(context);
        },
        child: Text('Try Again',
        style: TextStyle(fontSize: 18),
        ),
        style: ElevatedButton.styleFrom(
        padding: EdgeInsets.symmetric(horizontal: 20, vertical: 15), // Adjust the padding as needed
        ),
        ),
        SizedBox(height: 20),
        ElevatedButton(
            onPressed: () {
            // Implement exit logic
        },
        child: Text('Exit',
        style: TextStyle(fontSize: 18),
        ),
        style: ElevatedButton.styleFrom(
        padding: EdgeInsets.symmetric(horizontal: 20, vertical: 15), // Adjust the padding as needed
        ),
        ),
        ],
        ),
        ),
        );
    }
}

File: counting_and_sequencing/source_code/lib/screens/one_to_ten_page.dart
import 'package:flutter/material.dart';
import '../utils/number_generator.dart';
import '../widgets/number_option_button.dart';
import '../widgets/non_clickable_number.dart';
import 'right_answer.dart';
import 'wrong_answer.dart';
class OneToTenPage extends StatefulWidget {
  final int initialScore;

  OneToTenPage({this.initialScore = 0});

  @override
  _OneToTenPageState createState() => _OneToTenPageState();
}

class _OneToTenPageState extends State<OneToTenPage> {
  bool showSpanish = false;
  List<String> sequence = [];
  String correctOption = '';
  List<String> options = [];
  late int score;

  @override
  void initState() {
    super.initState();
    score = widget.initialScore; // Initialize score from the widget's property
    generateSequence();
  }

  void generateSequence() {
    sequence = NumberGenerator.generateSequence(4);
    correctOption = NumberGenerator.generateCorrectOption(sequence);
    options = NumberGenerator.generateOptions(sequence);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        backgroundColor: Colors.purple[100],
        appBar: AppBar(
          title: Text('LEVEL 1'),
          centerTitle: true,
          backgroundColor: Colors.purple,
        ),
        body: Stack(children: [
          Positioned.fill(
            child: Image.asset(
              'assets/p5.gif', // Ensure your image path is correct
              fit: BoxFit.cover,
            ),
          ),
          Center(
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: [
                SizedBox(height: 20),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                  children: [
                    for (String number in sequence)
                      NonClickableNumber(
                          text: showSpanish
                              ? NumberGenerator.convertToSpanish(number)
                              : number),
                  ],
                ),
                SizedBox(height: 32),
                Text(
                  'Choose an Option',
                  style: TextStyle(
                    fontSize: 28,
                    fontWeight: FontWeight.bold,
                    color: Colors.white,
                  ),
                ),
                SizedBox(height: 20),
                for (String option in options)
                  NumberOptionButton(
                    text: showSpanish
                        ? NumberGenerator.convertToSpanish(option)
                        : option,
                    onPressed: () {
                      if (option == correctOption) {
                        setState(() {
                          score += 10; // Add points for a correct answer
                        });
                        navigate(context, true);
                      } else {
                        setState(() {
                          score -= 5; // Subtract points for a wrong answer
                        });
                        navigate(context, false);
                      }
                    },
                  ),
                SizedBox(height: 32),
                ElevatedButton(
                  onPressed: () {
                    setState(() {
                      showSpanish = !showSpanish;
                    });
                  },
                  child: Text(
                    showSpanish ? 'English' : 'Español',
                    style: TextStyle(fontSize: 23),
                  ),
                ),
              ],
            ),
          ),
        ]));
  }

  void navigate(BuildContext context, bool isCorrect) {
    Navigator.pushReplacement(
      context,
      MaterialPageRoute(
        builder: (context) => isCorrect
            ? RightAnswerPage(
            score: score,
            onNextQuestionPressed: () {
              Navigator.pushReplacement(
                  context,
                  MaterialPageRoute(
                      builder: (context) =>
                          OneToTenPage(initialScore: score)));
            })
            : WrongAnswerPage(score: score),
      ),
    );
  }
}